### PR TITLE
Fail early if the MCS CRDs are not available

### PR DESCRIPTION
**Pull request by skitt**
Explicitly list ServiceExports and ServiceImports, to make sure that all named contexts are reachable and have the necessary CRDs.

Accumulate all errors in setupClients() so that the user knows about errors in all named contexts instead of just the first one.

### CODE REVIEW

File: conformance/conformance_suite.go [hunk 1]
```diff
@@ -106,22 +109,42 @@ func setupClients() error {
...
+		if _, err := mcsClient.MulticlusterV1alpha1().ServiceExports("").List(context.TODO(), metav1.ListOptions{}); err != nil {
+			accumulatedErrors = append(accumulatedErrors,
+				fmt.Errorf("error listing ServiceExports on context %s, is the MCS API installed? %w", name, err))
+			continue
+		}
+
+		if _, err := mcsClient.MulticlusterV1alpha1().ServiceImports("").List(context.TODO(), metav1.ListOptions{}); err != nil {
+			accumulatedErrors = append(accumulatedErrors,
+				fmt.Errorf("error listing ServiceImports on context %s, is the MCS API installed? %w", name, err))
+			continue
```
Comment (tpantelis) [inline conformance/conformance_suite.go:137]: To support multiple version in the future, we'll likely retrieve the CRD resources but this is fine for now.

File: conformance/conformance_suite.go [hunk 2]
```diff
@@ -106,22 +109,42 @@ func setupClients() error {
...
 
 		clients[i] = clusterClients{name: name, k8s: k8sClient, mcs: mcsClient, rest: restConfig}
 	}
-	return nil
+
+	if len(accumulatedErrors) == 0 {
+		return nil
+	} else {
+		return errors.Join(accumulatedErrors...)
+	}
```
[the remark this answers is not in the document]
Comment (skitt) [reply conformance/conformance_suite.go:147]: So it does, and I’ve used that feature before…

File: conformance/conformance_suite.go [hunk 3]
```diff
@@ -83,45 +83,62 @@ var _ = BeforeSuite(func() {
...
-		}
+			clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &overrides)
 
-		name := context
-		if name == "" {
-			name = rawConfig.CurrentContext
-		}
+			rawConfig, err := clientConfig.RawConfig()
+			if err != nil {
+				return fmt.Errorf("error setting up a Kubernetes API client on context %s: %w", kubeContext, err)
```
Comment (tpantelis) [inline conformance/conformance_suite.go:97]: `errors.Wrap` is a bit simpler. Same below.

```suggestion
				return errors.Wrapf(err, "error setting up a Kubernetes API client on context %q", kubeContext)
```

Comment (skitt) [reply conformance/conformance_suite.go:97]: I know, but it’s not _that_ much simpler and we already have a different `errors` import…

**Pull request merged**